### DISCUSSION TITLE
Is there a way to use velero to backup the cluster and volumes to local filesystem instead of cloud?

### DISCUSSION THREAD

**Discussion by SarthakNarayan** [vmware-tanzu/velero#6898] [Community support Q&A] [answered]
Hi,

I am using Velero to back up my k3s cluster to AWS s3. I love the simplicity Velcro provides in terms of creating backups and restoring them.
Due to some requirements, I now want to back up my cluster and the pod volumes to the local filesystem instead of S3. Is it possible to do this? If yes then how?

Thank you.

**Comment by shubham-pampattiwar** [comment, marked as answer]
@SarthakNarayan Unfortunately, Velero as of now only supports object storage as a backup storage location target.

**Comment by draghuram** [comment]
@SarthakNarayan There do exist some workarounds such as running MinIO locally on top of the file system, if that is an option. Or you may want to consider this plug-in:
    https://github.com/replicatedhq/local-volume-provider

Note that the plug-in is not officially maintained by Velero maintainers.

**Comment by farazkhawaja** [comment]
Do we have an update on this? does velero support local system as backup storage location now? @sseago @shubham-pampattiwar

**Reply by kaovilai** [reply]
no.
you can try use https://github.com/replicatedhq/local-volume-provider

**Reply by farazkhawaja** [reply]
@kaovilai  any thoughts on when we might see support for using the local filesystem as a backup location? The idea would be to generate the backup tar file locally, which will then allow us to upload the tar onto our desired location such as an SFTP server.

**Reply by kaovilai** [reply]
given current bandwidth, my personal opinion (not a velero maintainer), is mid 2025 for manifests only, late 2025 for volume support.

https://github.com/replicatedhq/local-volume-provider already provide some of the functionality requested here, just not in an official velero project yet.

**Reply by kaovilai** [reply]
https://github.com/vmware-tanzu/velero/issues/1229 is the tracking issue.

**Reply by farazkhawaja** [reply]
any updates @kaovilai

**Reply by kaovilai** [reply]
No.

**Comment by Lyndon-Li** [comment]
Please vote on issue #1229 and share your use case of why NFS is required over minio
